gprofiler/spark/collector.py

```diff
@@ -163,47 +168,28 @@ def _spark_stage_metrics(self, running_apps: Dict[str, Tuple[str, str]]) -> Iter
         Get metrics for each Spark stage.
         """
         for app_id, (app_name, tracking_url) in running_apps.items():
+            tags = {"app_name": str(app_name), "app_id": str(app_id)}
+            logger.debug("Gathering stage metrics for app %s", app_id)
```

Comment:
Prefer using `extra` field instead of putting fields in the message:
```suggestion
            logger.debug("Gathering stage metrics for app", app_id=app_id)
```